### DISCUSSION TITLE
Configuração do VS Code para usar o VueJs

### DISCUSSION THREAD

**Discussion by Murillofilho86** [frontendbr/forum#1107] [Perguntas e Respostas]
Fala pessoal! 
Estou fazendo algumas coisas com o Vue e usando o VS Code, mas estou sem nenhuma extensão para indentação dos meus templates... Customizei um atalho para indentar todos os meus arquivos, html, javascript, css, C#, Node... Mas não está funcionando no Vue... Procurei na internet e achei o vue-snippets, instalei e mesmo assim não esta funcionando... Alguém tem uma dica pra ajudar o irmão aqui?

**Comment by bizarro** [comment]
Opa, o [Vetur](https://github.com/vuejs/vetur) não resolve tudo para você? Geralmente eu só uso ele quando preciso mexer em projetos Vue.

**Comment by marceloavf** [comment]
Recomendo esse pack da Sarah Drasner: https://marketplace.visualstudio.com/items?itemName=sdras.vue-vscode-extensionpack

**Comment by klarkc** [comment]
Além disso vc pode utilizar e extensão [eslint](https://github.com/Microsoft/vscode-eslint) e usar o [plugin](https://github.com/vuejs/eslint-plugin-vue) do Vue para eslint

**Comment by caioincau** [comment]
Uso o Vetur e estou super satisfeito.

**Comment by WillianSugiyama** [comment]
Cara, o Vuejs Extension Pack é o melhor pack pra isso, recomendo demais !

**Comment by Murillofilho86** [comment]
Instalei o Vue VS Code Extension Pack, tem o vetur incluso, é completíssimo... Mas não resolveu, o atalho que eu uso pra formatar ainda não funciona em todo código... Só formata o que esta dentro da tag script, porém os componentes, eu tenho que indentar manualmente. :(

**Comment by WillianSugiyama** [comment]
@Murillofilho86, você vai precisar adicionar uma linha na sua configuração de preferencia: 

![image](https://user-images.githubusercontent.com/16587394/40996451-2bda81b6-68d8-11e8-9fda-581ce5504fbd.png)
Vai ajudar na indentação.

**Comment by houstondapaz** [comment]
@Murillofilho86 adiciona estás configurações no seu use settings
`` 
    "vetur.format.defaultFormatter.html": "js-beautify-html",
    "prettier.singleQuote": true,
    "prettier.semi": false,
    "eslint.options": {
        "extensions": [
            ".js",
            ".vue"
        ]
    }
``

**Comment by plcosta** [comment]
Recomendo esse artigo do @danielschmitz

[http://vuejs-brasil.com.br/deixe-o-eslint-trabalhar-para-voce-no-visual-studio-code/](http://vuejs-brasil.com.br/deixe-o-eslint-trabalhar-para-voce-no-visual-studio-code/)

Tem sugestões de extensões e configuração do Eslint.
